atbash_cipher.cpp

```diff
@@ -2,12 +2,44 @@
 
 namespace atbash_cipher {
 
-std::string encode(const std::string& /*text*/) {
-    return "";
+static constexpr size_t GROUP_SIZE = 5;
+
+inline bool should_insert_space(const std::string& encoded) {
+    return (encoded.length() % (GROUP_SIZE + 1) == GROUP_SIZE);
+}
+
+inline char atbash_transform(char c) {
+    if (std::isalpha(c)) {
+        return 'z' - (std::tolower(c) - 'a');
+    }
+    return c;
+}
+
+std::string encode(const std::string& plain_text) {
+    std::string encoded;
+
+    for (char c : plain_text) {
+        if (std::isalnum(c)) {
+            if (should_insert_space(encoded)) {
+                encoded += ' ';
+            }
+            encoded += atbash_transform(c);
+        }
+    }
```

Comment:
Consider using early return to avoid the arrow-code pattern (deep nesting of the logic).
```suggestion
    for (char c : plain_text) {
        if (!std::isalnum(c))
            continue;
            
        if (should_insert_space(encoded)) {
            encoded += ' ';
        }
        encoded += atbash_transform(c);
    }
```

Reply:
Fixed.